src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -8541,6 +8541,126 @@ public void RefFieldsConsideredManaged_Indirect()
             Assert.True(comp.GetTypeByMetadataName("StructWithIndirectRefField").IsManagedTypeNoUseSiteDiagnostics);
         }
 
+        [WorkItem(62618, "https://github.com/dotnet/roslyn/issues/62618")]
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void RefAssignValueScopeMismatch_01(LanguageVersion languageVersion)
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        Span<int> s = new[] { 1 };
+        M(ref s);
+    }
+    static void M(ref Span<int> s)
+    {
+        Span<int> local = stackalloc int[] { 1 };
+        ref Span<int> rL = ref local;
+        rL = ref s; // 1
```

Comment:
Could we also do `s = ref rL`? Would that hit the parameter case you were looking for?

Reply:
`s = ref rL;` hits the first code path in `BindAssignment()` because the _ref-safe-to-escape_ of `s` is wider than the _ref-safe-to-escape_ of `rL`.